src/mongo/util/net/listen.cpp

```diff
@@ -51,6 +52,23 @@
 
 #endif
 
+// Make sure our Linux has the timer we need
+#if defined(__linux__)
+    #ifndef CLOCK_MONOTONIC_COARSE
+        #error "Your ancient kernel doesn't support CLOCK_MONOTONIC_COARSE: upgrade to v2.6.31 or newer"
```

Comment:
Thanks @erh! Really appreciate it. It's very helpful to know what we're targeting.

@jcalvinowens it looks like since we have to target 2.6.18 that CLOCK_PROCESS_CPUTIME_ID, CLOCK_THREAD_CPUTIME_ID, CLOCK_REALTIME, and CLOCK_MONOTONIC are available according to [the man page](http://man7.org/linux/man-pages/man2/clock_gettime.2.html). Is knowing that we have to target 2.6.18 enough to unblock you? There's a chance I might be able to look at this, but I'm very unexperienced with it, so if you would be able to comment here as to an approach that you think would work, that would be really helpful.